pypif_sdk/interop/mdf.py

```diff
@@ -0,0 +1,229 @@
+from pypif_sdk.readview import ReadView
+from pypif.obj.common import Value, ProcessStep
+from citrination_client import PifQuery, DatasetQuery, SystemQuery, Filter
+from ..util.citrination import get_client
+from pypif.pif import dumps
+
+import re
+
+
+def query_to_mdf_records(query=None, dataset_id=None, acl=None):
+    """Evaluate a query and return a list of MDF records
+
+    If a datasetID is specified by there is no query, a simple
+    whole dataset query is formed for the user
+    """
+    if not query and not dataset_id:
+        raise ValueError("Either query or datasetID must be specified")
+    if not query:
+        query = PifQuery(include_datasets=[dataset_id])
+    client = get_client()
+    if not client:
+        raise ValueError("Unable to create a citrination client; is 'CITRINATION_API_KEY' in the env?")
```

Comment:
What about raising in the `get_client` method? That way you have a bit more control over the error message that you print.

Reply:
Doing that and having code that is OK with `get_client` not getting a client wrap it in a try block is basically the same.  I like this pattern.

Reply:
You're right though; there's more info further down.  Changed.